developer_docs/public_api.md

```diff
@@ -0,0 +1,433 @@
+# Public API
+
+⚠️: This is currently a proposal, not an API that exists.
+
+# Authentication
+
+- Access to the API is available via a Personal Access Token, linked to an existing editor user account
+- Created via a logged-in user’s Settings page
+- Tokens can be deleted via the Settings page
+- Tokens are presented to the API using basic auth
+- Tokens can also be sent via the `Authorization: Bearer {your_access_token}` HTTP header
+
+# API Access
+
+- API writes are rate-limited to X per second, per access token
+- Operations are transactional, e.g. if a file is somehow invalid, the sketch won’t be left partially uploaded
+- All requests send and receive `Content-Type: application/json` unless otherwise stated
+
+# Models
+
+The API accepts and returns the following model objects, as JSON.
+
+## Sketch
+
+| Name  |       Type        |                                     Description                                      |
+| ----- | ----------------- | ------------------------------------------------------------------------------------ |
+| name  | String            | The sketch’s title                                                                   |
+| files | DirectoryContents | The files and directories in this sketch. See `DirectoryContents` for the structure. |
```

Comment:
The more I think about it, I'm not sure this should be called `files` since it can contain both files and directories. I can't think of a better name though.

- `contents` but then it's easily typo'd with `File.content`
- `folder`
- `items`

Any ideas @catarak?

Reply:
hmmmm! i think my reasoning behind `files` is the unix reasoning behind files, where both directories and text/media files are files. out of the ones you've listed, `files` to me is still the one i like the most.